sklearn/utils/estimator_checks.py

```diff
@@ -4269,29 +4274,31 @@ def _output_from_fit_transform(transformer, name, X, df, y):
         ("fit.transform/array/df", X, df),
         ("fit.transform/array/array", X, X),
     ]
-    for (
-        case,
-        data_fit,
-        data_transform,
-    ) in cases:
-        transformer.fit(data_fit, y)
-        if name in CROSS_DECOMPOSITION:
-            X_trans, _ = transformer.transform(data_transform, y)
-        else:
-            X_trans = transformer.transform(data_transform)
-        outputs[case] = (X_trans, transformer.get_feature_names_out())
+    if all(hasattr(transformer, meth) for meth in ["fit", "transform"]):
```

Comment:
Is it possible to not have ``fit`` ?

Reply:
It would really be a corner case: `BaseEstimator` does not impose to have `fit` and a stateless estimator does not impose to call `fit` before `transform`. So I could imagine that we would allow a stateless estimator inheriting from `BaseEstimator` to not implement `fit`.

Reply:
(and I don't like it :))

Reply:
We don't have that in scikit-learn, so let's say we keep it safe for third party estimators